test/extensions/filters/network/thrift_proxy/translation_integration_test.cc

```diff
@@ -0,0 +1,195 @@
+#include "envoy/config/filter/network/thrift_proxy/v2alpha1/thrift_proxy.pb.h"
+
+#include "extensions/filters/network/well_known_names.h"
+
+#include "test/extensions/filters/network/thrift_proxy/integration.h"
+#include "test/test_common/network_utility.h"
+
+#include "absl/strings/ascii.h"
+#include "gtest/gtest.h"
+
+using testing::Combine;
+using testing::TestParamInfo;
+using testing::TestWithParam;
+using testing::Values;
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ThriftProxy {
+namespace {
+std::string thrift_config;
+} // namespace
+
+class ThriftTranslationIntegrationTest
+    : public BaseThriftIntegrationTest,
+      public TestWithParam<std::tuple<TransportType, ProtocolType, TransportType, ProtocolType>> {
+public:
+  ThriftTranslationIntegrationTest() : BaseThriftIntegrationTest(thrift_config) {}
+
+  static void SetUpTestCase() {
+    thrift_config = ConfigHelper::BASE_CONFIG + R"EOF(
+    filter_chains:
+      filters:
+        - name: envoy.filters.network.thrift_proxy
+          config:
+            stat_prefix: thrift_stats
+            route_config:
+              name: "routes"
+              routes:
+                - match:
+                    method_name: "add"
+                  route:
+                    cluster: "cluster_0"
+      )EOF";
+  }
+
+  void initialize() override {
+    TransportType downstream_transport, upstream_transport;
+    ProtocolType downstream_protocol, upstream_protocol;
+    std::tie(downstream_transport, downstream_protocol, upstream_transport, upstream_protocol) =
+        GetParam();
+
+    envoy::config::filter::network::thrift_proxy::v2alpha1::TransportType upstream_transport_proto;
+    switch (upstream_transport) {
```

Comment:
Factor these switches out to utility functions in a common library? Seems generally useful across tests, etc.